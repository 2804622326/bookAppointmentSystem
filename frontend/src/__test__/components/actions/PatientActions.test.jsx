import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import PatientActions from '../../../components/actions/PatientActions.jsx';

const appointment = {
  id: 1,
  appointmentDate: '2024-01-01',
  appointmentTime: '10:00',
  reason: 'Checkup',
};

test('renders PatientActions and handles interactions', () => {
  const onCancel = jest.fn();
  const onUpdate = jest.fn();
  const { asFragment } = render(
    <PatientActions
      onCancel={onCancel}
      onUpdate={onUpdate}
      isDisabled={false}
      appointment={appointment}
    />
  );

  const updateButton = screen.getByRole('button', { name: /update appointment/i });
  const cancelButton = screen.getByRole('button', { name: /cancel appointment/i });

  expect(updateButton).toBeInTheDocument();
  expect(cancelButton).toBeInTheDocument();

  fireEvent.click(cancelButton);
  expect(onCancel).toHaveBeenCalledWith(1);

  fireEvent.click(updateButton);
  expect(screen.getByText(/save update/i)).toBeInTheDocument();

  expect(asFragment()).toMatchSnapshot();
});
