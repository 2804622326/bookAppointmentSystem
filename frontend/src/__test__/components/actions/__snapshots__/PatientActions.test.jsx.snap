// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PatientActions and handles interactions 1`] = `
<DocumentFragment>
  <section
    class="d-flex justify-content-end gap-2 mt-2 mb-2"
  >
    <button
      class="btn btn-warning btn-sm"
      type="button"
    >
      Update Appointment
    </button>
    <button
      class="btn btn-danger btn-sm"
      type="button"
    >
      Cancel Appointment
    </button>
  </section>
</DocumentFragment>
`;
